app/models/screened_ip_address.rb

```diff
@@ -67,7 +78,8 @@ def self.match_for_ip_address(ip_address)
     #
     #   http://www.postgresql.org/docs/9.1/static/datatype-net-types.html
     #   http://www.postgresql.org/docs/9.1/static/functions-net.html
-    order('masklen(ip_address) DESC').find_by("? <<= ip_address", ip_address.to_s)
+    ip_address = IPAddr === ip_address ? ip_address.to_cidr_s : ip_address.to_s
```

Comment:
When is `ip_address` a string versus an `IPAddr`? 🤔

Reply:
It's a string usually when the input comes from a controller and an `IPAddr` when it was already parsed or comes from the DB. We used to call `to_s` always, but that is not good enough because sometimes a range is passed and `to_s` does not include the masklen:

```
pry(main)> IPAddr.new("192.168.2.15/24").to_s
=> "192.168.2.0"
```

Reply:
Might be worth adding a comment.

Reply:
I'd recommend converting into an IPAddr right at the very top and always handling it as such when possible rather than a string - it avoids surprises.